webapp/src/views/userSettings/notifications/SettingsRow.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Box, styled, Switch, Tooltip, Typography } from '@mui/material';
+import { components, operations } from 'tg.service/apiSchema.generated';
+import { useApiMutation } from 'tg.service/http/useQueryApi';
+import { messageService } from 'tg.service/MessageService';
+import { T } from '@tolgee/react';
+
+const StyledSwitch = styled(Box)`
+  text-align: center;
+`;
+
+type Props = {
+  description: string;
+  subdescription: string;
+  group: operations['putNotificationSetting']['parameters']['query']['group'];
+  channels: components['schemas']['NotificationSettingGroupModel'];
+  disabledInApp?: boolean;
+  disabledEmail?: boolean;
+  afterChange: () => void;
+};
+
+export const SettingsRow: React.FC<Props> = ({
+  description = '',
+  subdescription = '',
+  group,
+  channels,
+  disabledInApp = false,
+  disabledEmail = false,
+  afterChange = () => {},
+}: Props) => {
+  const saveMutation = useApiMutation({
+    url: '/v2/notifications-settings',
+    method: 'put',
+  });
+
+  const saveSettings = (
+    channel: operations['putNotificationSetting']['parameters']['query']['channel'],
+    enabled: boolean
+  ) => {
+    saveMutation.mutate(
+      {
+        query: {
+          group: group,
+          channel: channel,
+          enabled: enabled,
+        },
+      },
+      {
+        onSuccess() {
+          messageService.success(
+            <T keyName="User data - Successfully updated!" />
```

Comment:
This is not very nice key name.

Reply:
True, thanks for the catch.

Btw fun fact: I copy-pasted it from [an existing place](https://github.com/tolgee/tolgee-platform/blob/main/webapp/src/views/userSettings/userProfile/UserProfileView.tsx#L43).